cdap-examples/Purchase/src/test/java/co/cask/cdap/examples/purchase/PurchaseAppTest.java

```diff
@@ -62,7 +63,8 @@ public void test() throws TimeoutException, InterruptedException, IOException {
 
     try {
       // Wait for the last Flowlet processing 5 events, or at most 5 seconds
-      RuntimeMetrics metrics = RuntimeStats.getFlowletMetrics("PurchaseHistory", "PurchaseFlow", "collector");
+      RuntimeMetrics metrics = RuntimeStats.getFlowletMetrics(Constants.DEFAULT_NAMESPACE, "PurchaseHistory",
```

Comment:
Yes, I would in fact say that we should be testing with multiple namespaces in tests now, so don't want to hard-code it to ``DEFAULT_NAMESPACE`` always. 

For v2 APIs (in which most of this PR is), it could be ok to always presume that you are using ``DEFAULT_NAMESPACE``, but for v3 APIs (eventually all tests should migrate to use v3), we should specify different namespaces.

Thoughts?